src/xdgiconloader/xdgiconloader.cpp

```diff
@@ -787,121 +788,142 @@ QPixmap PixmapEntry::pixmap(const QSize &size, QIcon::Mode mode, QIcon::State st
+                  % filename
+                  % HexString<int>(mode)
+                  % HexString<int>(state)
+                  % HexString<int>(size.width())
+                  % HexString<int>(size.height());
+    if (!QPixmapCache::find(key, &pm))
+    {
+        int icnSize = qMin(size.width(), size.height());
+        pm = QPixmap(icnSize, icnSize);
+        pm.fill(Qt::transparent);
 
-    return QPixmap();
+        QSvgRenderer renderer;
+        if (renderer.load(filename))
+        {
+            QPainter p;
+            p.begin(&pm);
+            renderer.render(&p, QRect(0, 0, icnSize, icnSize));
+            p.end();
+        }
+
+        svgIcon = QIcon(pm);
+        if (QIconEngine *engine = svgIcon.data_ptr() ? svgIcon.data_ptr()->engine : nullptr)
+            pm = engine->pixmap(size, mode, state);
+        QPixmapCache::insert(key, pm);
+    }
+
+    return pm;
 }
 
 static const QString STYLE = QStringLiteral("\n.ColorScheme-Text, .ColorScheme-NeutralText {color:%1;}\
 \n.ColorScheme-Background {color:%2;}\
 \n.ColorScheme-Highlight {color:%3;}");
-// Note: Qt palette does not have any colors for positive/negative text
+// NOTE: Qt palette does not have any colors for positive/negative text
 // .ColorScheme-PositiveText,ColorScheme-NegativeText {color:%4;}
 
 QPixmap ScalableFollowsColorEntry::pixmap(const QSize &size, QIcon::Mode mode, QIcon::State state)
 {
     QPixmap pm;
-    // see ScalableEntry::pixmap() for the reason
-    if (QIconEngine *engine = svgIcon.data_ptr() ? svgIcon.data_ptr()->engine : nullptr)
-        pm = engine->pixmap(size, mode, state);
+    if (size.isEmpty())
+        return pm;
 
-    // Note: not checking the QIcon::isNull(), because in Qt5.10 the isNull() is not reliable
-    // for svg icons desierialized from stream (see https://codereview.qt-project.org/#/c/216086/)
-    if (pm.isNull())
+    const QPalette pal = qApp->palette();
+    QString txtCol, bgCol, hCol;
+    if (mode == QIcon::Disabled)
+    {
+        txtCol = pal.color(QPalette::Disabled, QPalette::WindowText).name();
+        bgCol = pal.color(QPalette::Disabled, QPalette::Window).name();
+        hCol = pal.color(QPalette::Disabled, QPalette::Highlight).name();
+    }
+    else
     {
-        // The following lines are adapted and updated from KDE's "kiconloader.cpp" ->
-        // KIconLoaderPrivate::processSvg() and KIconLoaderPrivate::createIconImage().
-        // They read the SVG color scheme of SVG icons and give images based on the icon mode.
-        QHash<int, QByteArray> svg_buffers;
+        if (mode == QIcon::Selected)
+        {
+            txtCol = pal.highlightedText().color().name();
+            bgCol = pal.highlight().color().name();
+        }
+        else // normal or active
+        {
+            txtCol = pal.windowText().color().name();
+            bgCol = pal.window().color().name();
+        }
+        hCol = pal.highlight().color().name();
+    }
+    QString key = QLatin1String("lxqt_")
```

Comment:
Just suggestion... wouldn't it be better to to use the same key structure for both `ScalableFollowsColorEntry` & `ScalableEntry`? Or be the one at least prefix of the extended?

Reply:
The only difference is that, in `ScalableFollowsColorEntry`, we need three colors (not the whole palette) but we don't need them in `ScalableEntry`.

BTW, what do you think about increasing `QPixmapCache` limit? I didn't think that was needed but wanted to ask your opinion.

Reply:
> but we don't need them in ScalableEntry.

Sure... but then wouldn't it be better to have the same fixed-width prefix... like:
```cpp
      QString key = QLatin1String("lxqt_")
                    % filename
                    % HexString<int>(mode)
                    % HexString<int>(state)
                    % HexString<int>(size.width())
                    % HexString<int>(size.height())
                    % txtCol % bgCol % hCol;
```

..and can't the concatenation of *Col allow a collision?

> what do you think about increasing QPixmapCache limit?

Never looked into that. But I we need to increase...then I would vote for allowing it to be configurable.

Reply:
> but then wouldn't it be better to have the same fixed-width prefix... like:

No objection from me if you prefer it. No collision will happen either. Will do it soon.

> But if we need to increase...then I would vote for allowing it to be configurable

I agree. Haven't found a real use case yet; just an idea.

Reply:
Done; made the cache keys similar to each other.